.circleci/config.yml

```diff
@@ -35,32 +35,45 @@ commands:
     description: "Restore the cache with pyspec keys"
     steps:
       - restore_cached_venv:
-          venv_name: v7-pyspec
+          venv_name: v9-pyspec
           reqs_checksum: cache-{{ checksum "tests/core/pyspec/requirements.txt" }}-{{ checksum "tests/core/pyspec/requirements-testing.txt" }}
   save_pyspec_cached_venv:
     description: Save a venv into a cache with pyspec keys"
     steps:
       - save_cached_venv:
-          venv_name: v7-pyspec
+          venv_name: v9-pyspec
           reqs_checksum: cache-{{ checksum "tests/core/pyspec/requirements.txt" }}-{{ checksum "tests/core/pyspec/requirements-testing.txt" }}
           venv_path: ./tests/core/pyspec/venv
-  restore_deposit_contract_cached_venv:
-    description: "Restore the cache with deposit_contract keys"
+  restore_deposit_contract_compiler_cached_venv:
+    description: "Restore the venv from cache for the deposit contract compiler"
     steps:
       - restore_cached_venv:
-          venv_name: v9-deposit-contract
-          reqs_checksum: cache-{{ checksum "tests/core/pyspec/requirements.txt" }}-{{ checksum "deposit_contract/requirements-testing.txt" }}
-  save_deposit_contract_cached_venv:
-    description: Save a venv into a cache with deposit_contract keys"
+          venv_name: v15-deposit-contract-compiler
+          reqs_checksum: cache-{{ checksum "deposit_contract/compiler/requirements.txt" }}
+  save_deposit_contract_compiler_cached_venv:
+    description: "Save the venv to cache for later use of the deposit contract compiler"
     steps:
       - save_cached_venv:
-          venv_name: v9-deposit-contract
-          reqs_checksum: cache-{{ checksum "tests/core/pyspec/requirements.txt" }}-{{ checksum "deposit_contract/requirements-testing.txt" }}
-          venv_path: ./deposit_contract/venv
+          venv_name: v15-deposit-contract-compiler
+          reqs_checksum: cache-{{ checksum "deposit_contract/compiler/requirements.txt" }}
+          venv_path: ./deposit_contract/compiler/venv
+  restore_deposit_contract_tester_cached_venv:
+    description: "Restore the venv from cache for the deposit contract tester"
+    steps:
+      - restore_cached_venv:
+          venv_name: v15-deposit-contract-tester
+          reqs_checksum: cache-{{ checksum "tests/core/pyspec/requirements.txt" }}-{{ checksum "deposit_contract/tester/requirements.txt" }}
+  save_deposit_contract_tester_cached_venv:
+    description: "Save the venv to cache for later use of the deposit contract tester"
+    steps:
+      - save_cached_venv:
+          venv_name: v15-deposit-contract-tester
+          reqs_checksum: cache-{{ checksum "tests/core/pyspec/requirements.txt" }}-{{ checksum "deposit_contract/tester/requirements.txt" }}
+          venv_path: ./deposit_contract/tester/venv
```

Comment:
Not due to this PR, but I recalled that sometimes we needed to kick the`deposit_contract_cached_venv` cache by updating `venv_name` version manually. It's probably because `deposit-contract-tester` depends on `pyspec`, but we couldn't get **the `pyspec` version** in the checksum.

Will we have an explicit ` pyspec` package version in #1584?

Reply:
Although packaging works nicely, and it can be pulled in as git dependency with pip, I'm not sure about versioning of the pyspec itself yet. Putting in the spec version could work, but staying on 0.0.0, with git as versioning, serves all current use cases for now fine.